### PR TITLE
Implement basic require method

### PR DESCRIPTION
**Pull request by hmcguire-shopify**
### Summary

Before this change, calls to require were going to the host ruby
Kernel#require, which meant that we weren't actually running any of the
code ourselves.

### CODE REVIEW

File: lib/yarv.rb [hunk 1]
```diff
@@ -118,6 +135,22 @@ def with_frame(iseq)
         @stack = @stack[0..current_stack_length]
       end
     end
+
+    # Pushes a new frame onto the stack, executes the instructions contained
+    # within this instruction sequence, then pops the frame off the stack.
+    def eval(iseq)
```
Comment (hmcguire-shopify) [inline lib/yarv.rb:141]: I _think_ eval belongs on the EC because we will need to be able to compile and run new InstructionSequences in the same context.

Comment (kddnewton) [reply lib/yarv.rb:141]: I'm fine with this change moving it to the EC but can you leave the `eval` method here and just make it `eval(context = ExecutionContext.new) = context.eval(self)` to keep that API? That way it still mirrors the `RubyVM::InstructionSequence` one

**Review by kddnewton — CHANGES_REQUESTED** [review]
Just a couple of comments

File: lib/yarv.rb [hunk 2]
```diff
@@ -12,6 +12,23 @@ class Main
...
+        next unless File.file?(file_path) && File.readable?(file_path)
+
+        iseq =
+          File.open(file_path, "r") do |f|
+            YARV.compile(f.read, file_path, file_path)
+          end
+      end
+
+      # TODO: run the iseq in the ExecutionContext. Currently blocked on some
+      # missing instructions needed for mspec-run
```
[the remark this answers is not in the document]
Comment (hmcguire-shopify) [reply lib/yarv.rb:30]: Ah, maybe I should have worded the message differently: currently instantiating the InstructionSequence errors so it doesn't even get to a point where its runnable yet.

[the remark this answers is not in the document]
Comment (kddnewton) [reply lib/yarv.rb:30]: Yeah that's fine. Go ahead and add the code to run and just let it fail.

**Pull request merged**